tests/misc-utils.h

```diff
@@ -0,0 +1,96 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright 2016 Kinvolk GmbH
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Author: Krzesimir Nowak <krzesimir@kinvolk.io>
+ */
+
+#pragma once
+
+#include <gio/gio.h>
+
+G_BEGIN_DECLS
+
+gboolean rm_rf (GFile *topdir,
+		GError **error);
+
+static inline GPtrArray *
+string_array_new (void)
+{
+  return g_ptr_array_new_with_free_func (g_free);
+}
+
+static inline GPtrArray *
+object_array_new (void)
+{
+  return g_ptr_array_new_with_free_func (g_object_unref);
+}
+
+static inline gpointer
+save_for_out_of_scope_gc (gpointer to_save, GPtrArray* saved)
+{
+  g_ptr_array_add (saved, to_save);
+  return to_save;
+}
+
+/* expects the out_of_scope_gc_strs GPtrArray variable to be defined
+ * in scope, use SSDEF */
+#define SS(s) (gchar *)save_for_out_of_scope_gc (s, out_of_scope_gc_strs)
+#define SSDEF g_autoptr(GPtrArray) out_of_scope_gc_strs = string_array_new ()
+
+/* expects the out_of_scope_gc_objs GPtrArray variable to be defined
+ * in scope, use OSDEF */
+#define OS(o) save_for_out_of_scope_gc (o, out_of_scope_gc_objs)
+#define OSDEF g_autoptr(GPtrArray) out_of_scope_gc_objs = object_array_new ()
```

Comment:
After having seen how this works out in the tests, I'm don't think this is a good idea from a readability perspective. I think it is worth trading the terseness gained from this approach with an approach which is verbose (e.g., declare g_autoptr` for the strings) but more readable.